### PR TITLE
arc_unpacker: fix build

### PR DESCRIPTION
**Pull request by futile**
###### Motivation for this change

Fix build of the package, as it was failing before:

https://hydra.nixos.org/build/141997371/log

```console
[...]
build/source/src/algo/crypt/lcg.cc: In constructor 'au::algo::crypt::Lcg::Lcg(au::algo::crypt::LcgKind, au::u32)':
/build/source/src/algo/crypt/lcg.cc:70:24: error: 'logic_error' is not a member of 'std'
   70 |             throw std::logic_error("Unknown LCG kind");
      |                        ^~~~~~~~~~~
make[2]: *** [CMakeFiles/libau.dir/build.make:160: CMakeFiles/libau.dir/src/algo/crypt/lcg.cc.o] Error 1
[...]
```

Also submitted as a PR to the upstream repo: https://github.com/vn-tools/arc_unpacker/pull/194

ZHF: #122042

@jonringer (ping meant for @NixOS/nixos-release-managers)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by futile** [comment]
@midchildan pinging you as you are listed as maintainer.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123091` at e780a610 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>arc_unpacker</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/archivers/arc_unpacker/default.nix:38:3:
```
   |
38 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`checkPhase` should probably contain `runHook preCheck` and `runHook postCheck`.

Near pkgs/tools/archivers/arc_unpacker/default.nix:32:3:
```
   |
32 |   checkPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/tools/archivers/arc_unpacker/default.nix:51:5:
```
   |
51 |     license = licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

**Comment by futile** [comment]
Added a commit that fixes the bot suggestions.

**Comment by midchildan** [comment]
Thanks! Could you also apply the following to fix the Darwin build?

```diff
diff --git a/pkgs/tools/archivers/arc_unpacker/default.nix b/pkgs/tools/archivers/arc_unpacker/default.nix
index c6259cf11b0..7e2ee60a156 100644
--- a/pkgs/tools/archivers/arc_unpacker/default.nix
+++ b/pkgs/tools/archivers/arc_unpacker/default.nix
@@ -1,5 +1,17 @@
-{ lib, stdenv, fetchFromGitHub, cmake, makeWrapper, boost, libpng, libjpeg, zlib
-, openssl, libwebp, catch }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, makeWrapper
+, boost
+, libiconv
+, libjpeg
+, libpng
+, libwebp
+, openssl
+, zlib
+, catch
+}:
 
 stdenv.mkDerivation rec {
   pname = "arc_unpacker-unstable";
@@ -16,7 +28,8 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ cmake makeWrapper catch ];
-  buildInputs = [ boost libpng libjpeg zlib openssl libwebp ];
+  buildInputs = [ boost libpng libjpeg zlib openssl libwebp ]
+    ++ lib.optionals stdenv.isDarwin [ libiconv ];
 
   patches = [
     # Add a missing `<stdexcept>` import that caused the build to fail.
```

**Comment by jonringer** [comment]
> ```diff
> +  buildInputs = [ boost libpng libjpeg zlib openssl libwebp ]
> +    ++ lib.optionals stdenv.isDarwin [ libiconv ];
> ```

This can also be done in another PR. Trying to get as much merged in as I can before staging-next gets merged

**Comment by futile** [comment]
I can add it, or keep it out, which would be better?

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 123091 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>arc_unpacker</li>  </ul></details>

**Pull request merged**
